src/vuestic-theme/vuestic-components/va-pagination/VaPagination.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <div class="va-pagination">
+    <va-button-group>
+      <va-button
+        v-if="lastPage !== length"
+        outline color="info"
+        :disabled="disabled"
+        icon="fa fa-angle-double-left"
+        @click="changePage(1)"
+      />
+      <va-button
+        v-if="lastPage !== length"
+        outline color="info"
+        :disabled="disabled"
+        icon="fa fa-angle-left"
+        @click="changePage(value - 1)"
+      />
+      <va-button
+        outline color="info"
+        :disabled="disabled"
+        v-for="n in paginationRange" :key="n"
+        :class="{ 'va-button--active': n === value }"
+        @click="changePage(n)"
+      >
+        {{ n }}
+      </va-button>
+      <va-button
+        v-if="lastPage !== length"
+        outline color="info"
+        :disabled="disabled"
+        icon="fa fa-angle-right"
+        @click="changePage(value + 1)"
+      />
+      <va-button
+        v-if="lastPage !== length"
+        outline color="info"
+        :disabled="disabled"
+        icon="fa fa-angle-double-right"
+        @click="changePage(lastPage)"
+      />
+    </va-button-group>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'va-pagination',
+  props: {
+    value: {
+      type: Number,
+    },
+    length: {
+      type: Number,
+      default: 5
+    },
+    totalPages: {
+      type: Number
+    },
+    disabled: {
+      type: Boolean
+    }
+  },
+  methods: {
+    changePage (pageNum) {
+      this.$emit('input', pageNum)
+      this.$emit('change-page', pageNum)
```

Comment:
Isn't input enough?

Reply:
It doesn't enough. F.e., pagination component in vuetable also allows additional actions on changing value and it can be suitable not only for tables in future
See vuestic-datatable component as a reference

Reply:
Still good chance we have to remove this: `this.$emit('change-page', pageNum)`